src/pygeoprocessing/geoprocessing.py

```diff
@@ -3415,11 +3422,15 @@ def _convolve_2d_worker(
         signal_block = signal_band.ReadAsArray(**signal_offset)
         kernel_block = kernel_band.ReadAsArray(**kernel_offset)
 
-        if signal_nodata is not None and ignore_nodata:
-            # if we're ignoring nodata, we don't want to add it up in the
-            # convolution, so we zero those values out
-            signal_nodata_mask = numpy.isclose(signal_block, signal_nodata)
-            signal_block[signal_nodata_mask] = 0.0
+        if ignore_nodata:
```

Comment:
I think this `ignore_nodata` is fine because it's in the worker function, and should not be `ignore_nodata_and_edges` but wanted to check to make sure.

Reply:
Exactly, the meaning is different inside of the worker -- there's no edges there, only nodata.